Sofa/framework/Core/src/sofa/core/behavior/PartialLinearSolver.h

```diff
@@ -0,0 +1,53 @@
+/******************************************************************************
+*                 SOFA, Simulation Open-Framework Architecture                *
+*                    (c) 2006 INRIA, USTL, UJF, CNRS, MGH                     *
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#pragma once
+
+#include <sofa/core/config.h>
+#include <sofa/core/objectmodel/BaseObject.h>
+
+#include <list>
+
+namespace sofa::core::behavior
+{
+
+/**
+ *  \brief Interface describing partial linear solvers API
+ *
+ */
+class PartialLinearSolver : public virtual sofa::core::objectmodel::BaseObject
+{
+public:
+    SOFA_ABSTRACT_CLASS(PartialLinearSolver, objectmodel::BaseObject);
```

Comment:
You should add at least a key function (see https://github.com/sofa-framework/sofa/pull/3240). Add a default constructor. Its definition must be in a cpp file